octodns/provider/easydns.py

```diff
@@ -0,0 +1,453 @@
+        sess.headers.update({'accept': 'application/json'})
+        self._sess = sess
+
+    def _request(self, method, path, params=None, data=None):
+        url = '{}{}'.format(self.basepath, path)
+        resp = self._sess.request(method, url, params=params, json=data)
+        if resp.status_code == 400:
+            self.log.debug('Response code 400, path=%s', path)
+            if method == 'GET' and path[:8] == '/domain/':
+                raise EasyDNSClientNotFound()
+            raise EasyDNSClientBadRequest()
+        if resp.status_code == 401:
+            raise EasyDNSClientUnauthorized()
+        if resp.status_code == 403 or resp.status_code == 404:
+            raise EasyDNSClientNotFound()
+        resp.raise_for_status()
+        return resp
+
+    def domain(self, name):
+        path = '/domain/{}'.format(name)
+        return self._request('GET', path).json()
+
+    def domain_create(self, name):
+        # EasyDNS allows for new domains to be created for the purpose of DNS
+        # only, or with domain registration. This function creates a DNS only
+        # record expectig the domain to be registered already
+        path = '/domains/add/{}'.format(name)
+        domainData = {'service': 'dns',
+                      'term': 1,
+                      'dns_only': 1,
+                      'portfolio': self.domainPortfolio,
+                      'currency': self.defaultCurrency}
+        self._request('PUT', path, data=domainData).json()
+
+        # EasyDNS creates default records for MX, A and CNAME for new domains,
+        # we need to delete those default record so we can sync with the source
+        # records, first we'll sleep for a second before gathering new records
+        # We also create default NS records, but they won't be deleted
+        sleep(1)
+        records = self.records(name, True)
+        for record in records:
+            if record['type'] in ('A', 'MX', 'CNAME'):
+                self.record_delete(name, record['id'])
+
+    def records(self, zone_name, raw=False):
+        if raw:
+            path = '/zones/records/all/{}'.format(zone_name)
+        else:
+            path = '/zones/records/parsed/{}'.format(zone_name)
+
+        ret = []
+        resp = self._request('GET', path).json()
+        ret += resp['data']
+
+        # EasyDNS supports URL forwarding, stealth URL forwarding and DYNamic
+        # A records so we'll convert them to their underlying DNS record
+        # types before processing
+        for record in ret:
+            # change any apex record to empty string
+            if record['host'] == '@':
+                record['host'] = ''
+
+            # change any apex value to zone name
+            if record['rdata'] == '@':
+                record['rdata'] = '{}.'.format(zone_name)
+
+            # change "URL" & "STEALTH" to a "CNAME"
+            if record['type'] == "URL" or record['type'] == "STEALTH":
+                record['type'] = 'CNAME'
```

Comment:
If `URL` is for redirects to arbitrary URLs I'm not sure a `CNAME` will work here since it'd lose any path and scheme information from the URL. Other providers just ignore `URL` style records for this reason.

What are `STEALTH` records? Are they similar to `ALIAS` in other providers, i.e. the DNS resolvers do the lookup for the target and then return its value(s)?

Reply:
While EasyDNS's URL and STEALTH records do result in being sent to a specific URL, their underlying type is actually a CNAME (if you were to dig the DNS record it would return as a CNAME) which points to an EasyDNS host which then performs a redirect to the client's specified URL, so nothing is lost.  A STEALTH record is the same as a URL but the resulting URL is hidden from the browser (when possible).  

If you believe this would cause confusion with how other providers handle such records, I can ask the API author to change the record type during parsing so they come through as CNAME records and remove the related code from the provider class.

Reply:
Mostly concerned with what the `value` would be. This translation would result in a `CNAME` in the octoDNS config, but there's nothing that I see that would pull out the hostname part of that and record that in the config. 

Also there'd be no way afaict to edit these translated records. If you change the `CNAME` `value` in the config it'd try and push a `CNAME` `value` up to the API in a place that previously held a `STEALTH` or `URL` record. 

I believe octoDNS will need to ignore these types of records as it does for `URL` in other providers.

Reply:
No problem, I've updated the code to remove the URL and STEALTH record conversions.